Base/Implementations/Validation.cs

```diff
@@ -95,7 +93,15 @@ public IEnumerable GetErrors()
                 return new List<string>();
             }
             return _Errors;
+            
+        }
+
+        public List<IErrorMessage> GetErrorMessages()
+        {
+            Validate();
+            return _ErrorMessages;
```

Comment:
could this all be handled by doing a loop after validate on rulemap to dynamically create the error messages? this would reduce the class by not storing the error messages.

Reply:
I think I would prefer to get rid of the list of string errors and basically just replace it with a list of IErrorMessage and not have both. If there was a place where we just wanted the string messages that would be easy to get from the ErrorMessage.

Reply:
Will that screw up some bindings somewhere? It sounded like you had some WPF that was relying on that IEnumerable<string> Errors. I guess I could call mine Errors so at least the binding wouldn't totally break.

Reply:
It might. Do errors like you suggest